File: app/src/main/java/com/lostincontext/application/LostApplication.java
package com.lostincontext.application;

import android.app.Application;
import android.support.annotation.NonNull;


public class LostApplication extends Application {


    private ApplicationComponent applicationComponent;

    @Override public void onCreate() {
        super.onCreate();

        applicationComponent = DaggerApplicationComponent
                .builder()
                .applicationModule(new ApplicationModule(this))
                .build();

    }


    @NonNull
    public ApplicationComponent getAppComponent() {
        return applicationComponent;
    }

}

File: app/src/main/java/com/lostincontext/application/LostApplication.kt
package com.lostincontext.application

import android.app.Application


class LostApplication : Application() {


    lateinit var appComponent: ApplicationComponent
        private set

    override fun onCreate() {
        super.onCreate()

        appComponent = DaggerApplicationComponent.builder()
                .applicationModule(ApplicationModule(this))
                .build()

    }

}
